geode/src/main/java/org/apache/calcite/adapter/geode/rel/GeodeFilter.java

```diff
@@ -167,6 +167,7 @@ private String getLeftNodeFieldName(RexNode left) {
         // FIXME This will not work in all cases (for example, we ignore string encoding)
         return getLeftNodeFieldName(((RexCall) left).operands.get(0));
       case OTHER_FUNCTION:
+      case ITEM:
         return left.accept(new GeodeRules.RexToGeodeTranslator(this.fieldNames));
```

Comment:
Put `ITEM` in front of `OTHER_FUNCTION `.

Reply:
Is that for performance reasons ?  The JIT compiler should do a decent job of optimizing the switch-case for the more frequently occurring condition (assuming there's sufficient warm-up).

Reply:
No, just to make the logic more clear because `OTHER_FUNCTION` is a fallback.